File: AES256-OC/VaporClient/VaporClient/ViewController.m
//
//  ViewController.m
//  VaporClient
//
//  Created by luomeng on 2018/7/3.
//  Copyright © 2018年 tpf. All rights reserved.
//

#import "ViewController.h"
#import "NSData+AES.h"
#import "GTMBase64.h"
#import "AESEncryptor.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    NSString *aesKey = @"7854156156611111"; // 16位
    NSString *aesIV = nil; // 为nil时，自动填充为 0000000000000000
    
    NSString *link = @"https://www.google.com";
    NSString *encrypedLink = [AESEncryptor encrypt:link password:aesKey iv:aesIV];
    
    NSLog(@"encryped = %@", encrypedLink);
    
    NSString *plainTxt = [AESEncryptor decrypt:encrypedLink password:aesKey iv:aesIV];
    NSLog(@"plain = %@", plainTxt);
    
}

- (IBAction)sendAct:(id)sender {
    
}

- (IBAction)postAct:(id)sender {
}


@end

File: AES256-Swift/SwiftClient/SwiftClient/ViewController.swift
//
//  ViewController.swift
//  SwiftClient
//
//  Created by luomeng on 2019/3/1.
//  Copyright © 2019年 comp. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        let key = "7854156156611111"
        let iv =  "0000000000000000"
        let aes = try? AES(key: key, iv: iv, padding: .pkcs7)
        
        let plainTxt = "https://www.instagram.com"
        
        let cryptTxt = try? plainTxt.encryptToBase64(cipher: aes!)
        print(cryptTxt as Any)
        
        let decryptedLink = try? cryptTxt!!.decryptBase64ToString(cipher: aes!)
        print(decryptedLink as Any)
    }
    
    


}

